File: day-1-chronal-calibration/chronal-calibration.js
const fs = require('fs');

const frequencies = fs.readFileSync('data.txt').toString().split("\n").map((element) => parseInt(element)).filter((element) => !isNaN(element));

console.log(frequencies.reduce((a, b) => a + b, 0));

let currentFrequency = 0;
let seenFrequencies = new Set([currentFrequency]);
let count  = 0;
while (true) {
    frequencies.forEach((frequency) => {
        currentFrequency += frequency;
        if (seenFrequencies.has(currentFrequency)) {
            console.log(currentFrequency);
            process.exit(0);
        }
        seenFrequencies = seenFrequencies.add(currentFrequency);
    });
    count += 1;
    console.log(count, currentFrequency);
}

File: day-1-chronal-calibration/chronal-calibration.py
with open('data.txt', 'r') as f:
    frequencies = [int(line.strip()) for line in f.readlines()]

print(sum(frequencies))

current_frequency = 0
seen_frequencies = {current_frequency}
count = 0

while True:
    for frequency in frequencies:
        current_frequency += frequency
        if current_frequency in seen_frequencies:
            print(current_frequency)
            exit()
        seen_frequencies.add(current_frequency)
    count += 1
    print('pass {} {}'.format(count, current_frequency))
